pilot/pkg/xds/ads.go

```diff
@@ -613,7 +613,7 @@ func setTopologyLabels(proxy *model.Proxy) {
 		localityStr = proxy.ServiceInstances[0].Endpoint.Locality.Label
 	} else {
 		// If no service instances(this maybe common for a pure client), respect LocalityLabel
-		localityStr = proxy.Metadata.Labels[model.LocalityLabel]
+		localityStr = model.GetLocalityLabelOrDefault(proxy.Metadata.Labels[model.LocalityLabel], "")
```

Comment:
this seems to be no different? the changei s to provide a default, but the default is `""` which is already the default?

Reply:
The change is to [replace "." with "/"](https://github.com/istio/istio/blob/8f2e2dc5d57f6f1f7a453e03ec96ca72b2205783/pilot/pkg/model/service.go#L381-L382).

Reply:
This can not work, in this case the proxy.Metadata.Labels[model.LocalityLabel] is empty, and the result is "".